packages/webapp/__tests__/AccountNotificationsPage.tsx

```diff
@@ -0,0 +1,251 @@
+        visit={defaultVisit}
+        tokenRefreshed
+      >
+        <ProfileNotificationsPage />
+      </AuthContextProvider>
+    </QueryClientProvider>,
+  );
+};
+
+it('should accurately show device preference', async () => {
+  renderComponent([
+    createDevicePreferenceMock({ pushNotification: true }),
+    createGeneralPreferenceMock(),
+  ]);
+  await waitForNock();
+  const subscription = await screen.findByTestId('push_notification-switch');
+  expect(subscription).toHaveAttribute('data-testValue', 'true');
+});
+
+it('should change user push notification', async () => {
+  renderComponent();
+  await waitForNock();
+  let mutationCalled = false;
+  const pushNotification = true;
+  mockGraphQL({
+    request: {
+      query: UPDATE_DEVICE_PREFERENCE_MUTATION,
+      variables: {
+        data: { pushNotification },
+        deviceId: defaultVisit.sessionId,
+      },
+    },
+    result: () => {
+      mutationCalled = true;
+      return { data: { id: '' } };
+    },
+  });
+  mockGraphQL(createDevicePreferenceMock({ pushNotification: true }));
+  const subscription = await screen.findByTestId('push_notification-switch');
+  expect(subscription).not.toHaveAttribute('data-testValue');
+  await subscription.click();
+  await waitForNock();
+  const newSubscription = await screen.findByTestId('push_notification-switch');
+  expect(newSubscription).toHaveAttribute('data-testValue', 'true');
+  expect(mutationCalled).toBeTruthy();
+});
+
+it('should change user all email subscription', async () => {
+  renderComponent([
+    createDevicePreferenceMock(),
+    createGeneralPreferenceMock({
+      marketingEmail: false,
+      notificationEmail: true,
+    }),
+  ]);
+  await waitForNock();
+  let mutationCalled = false;
+  const data: GeneralNotificationPreference = {
+    marketingEmail: true,
+    notificationEmail: true,
+  };
+  mockGraphQL({
+    request: {
+      query: UPDATE_GENERAL_PREFERENCE_MUTATION,
+      variables: { data },
+    },
+    result: () => {
+      mutationCalled = true;
+      return { data: { id: '' } };
+    },
+  });
+  mockGraphQL(
+    createGeneralPreferenceMock({
+      marketingEmail: true,
+      notificationEmail: true,
+    }),
+  );
+  const subscription = await screen.findByTestId('email_notification-switch');
+  expect(subscription).not.toHaveAttribute('data-testValue');
+  await subscription.click();
+  await waitForNock();
+  const newSubscription = await screen.findByTestId(
+    'email_notification-switch',
+  );
+  expect(newSubscription).toHaveAttribute('data-testValue', 'true');
+  const marketingSubscription = await screen.findByTestId('marketing-switch');
+  expect(marketingSubscription).toHaveAttribute('data-testValue', 'true');
+  const notificationEmail = await screen.findByTestId('new_activity-switch');
+  expect(notificationEmail).toHaveAttribute('data-testValue', 'true');
+  expect(mutationCalled).toBeTruthy();
+});
+
+// when you comment all tests this work
```

Comment:
Why is this still here?

Reply:
Oh, tests issues. Lol. This is okay now 🤣